tlspuffin/src/openssl/mod.rs

```diff
@@ -167,25 +166,30 @@ impl Put<TLSProtocolBehavior> for OpenSSL {
         result
     }
 
-    fn reset(&mut self, _agent_name: AgentName) -> Result<(), Error> {
+    fn reset(&mut self, new_name: AgentName) -> Result<(), Error> {
+        self.config.descriptor.name = new_name;
+
+        #[cfg(feature = "claims")]
+        self.deregister_claimer();
```

Comment:
Not sure we want to remove al lclains from self!

Reply:
This only removes the claimer callback from the OpenSSL `ssl` object before performing the stream reset. Because when we do a hard reset the `ssl` object might be dropped in OpenSSL, it's safer to first retrieve and drop the callback allocated in Rust.

Reply:
You're right!